Dioxus native target doesn't compile

Thank you for providing this crate. It looks very promising.

I tried to use it in a Dioxus desktop app, but unfortunately, I can't compile it:
```
12:07:22 [cargo] error[E0599]: no method named `location` found for struct `web_sys::Window` in the current scope
  --> /home/me/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/table-rs-0.0.4/src/dioxus/table.rs:90:31
   |
90 |         let location = window.location();
   |                               ^^^^^^^^ method not found in `web_sys::Window`

12:07:22 [cargo] error[E0599]: no method named `location` found for struct `web_sys::Window` in the current scope
   --> /home/me/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/table-rs-0.0.4/src/dioxus/table.rs:100:27
    |
100 |         let href = window.location().href().unwrap();
    |                           ^^^^^^^^ method not found in `web_sys::Window`

12:07:22 [cargo] error[E0599]: no method named `history` found for struct `web_sys::Window` in the current scope
   --> /home/me/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/table-rs-0.0.4/src/dioxus/table.rs:107:14
    |
106 | /         window
107 | |             .history()
    | |_____________-^^^^^^^
    |
```

From the error message I assume, that certain functionality of `web_sys::Window` is not exposed for desktop targets, even though they run in a webview. Would there be any workaround or is this crate targeting `wasm` only?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,7 +18,7 @@ exclude = ["/assets", "/examples"]
 yew = { version = "0.21.0", default-features = false, optional = true }
 dioxus = { version = "0.7.1", optional = true }
 leptos = { version = "0.7.7", optional = true }
-web-sys = { version = "0.3.77", features = ["Window", "UrlSearchParams", "Url"]}
+web-sys = { version = "0.3.77", features = ["Window", "UrlSearchParams", "Url", "Location"]}
 gloo-timers = { version = "0.3.0", optional = true }
 
 [features]

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -18,7 +18,7 @@ exclude = ["/assets", "/examples"]
 yew = { version = "0.21.0", default-features = false, optional = true }
 dioxus = { version = "0.7.1", optional = true }
 leptos = { version = "0.7.7", optional = true }
-web-sys = { version = "0.3.77", features = ["Window", "UrlSearchParams", "Url", "Location"]}
+web-sys = { version = "0.3.77", features = ["Window", "UrlSearchParams", "Url", "Location", "History"]}
 gloo-timers = { version = "0.3.0", optional = true }
 
 [features]

--- a/src/dioxus/table.rs
+++ b/src/dioxus/table.rs
@@ -85,7 +85,7 @@ pub fn Table(props: TableProps) -> Element {
     let mut sort_order = use_signal(SortOrder::default);
     let mut search_query = use_signal(String::new);
 
-    #[cfg(target_arch = "wasm")]
+    #[cfg(target_family = "wasm")]
     use_effect(move || {
         let window = web_sys::window().unwrap();
         let location = window.location();
@@ -96,7 +96,7 @@ pub fn Table(props: TableProps) -> Element {
         }
     });
 
-    #[cfg(target_arch = "wasm")]
+    #[cfg(target_family = "wasm")]
     let update_search_param = move |query: &str| {
         let window = web_sys::window().unwrap();
         let href = window.location().href().unwrap();

--- a/src/dioxus/table.rs
+++ b/src/dioxus/table.rs
@@ -184,7 +184,7 @@ pub fn Table(props: TableProps) -> Element {
                         let val = e.value();
                         search_query.set(val.clone());
                         page.set(0);
-                        #[cfg(target_arch = "wasm")]
+                        #[cfg(target_family = "wasm")]
                         update_search_param(&val);
                     }
                 }

--- a/src/dioxus/table.rs
+++ b/src/dioxus/table.rs
@@ -85,6 +85,7 @@ pub fn Table(props: TableProps) -> Element {
     let mut sort_order = use_signal(SortOrder::default);
     let mut search_query = use_signal(String::new);
 
+    #[cfg(target_arch = "wasm")]
     use_effect(move || {
         let window = web_sys::window().unwrap();
         let location = window.location();
@@ -95,6 +96,7 @@ pub fn Table(props: TableProps) -> Element {
         }
     });
 
+    #[cfg(target_arch = "wasm")]
     let update_search_param = move |query: &str| {
         let window = web_sys::window().unwrap();
         let href = window.location().href().unwrap();
@@ -182,6 +184,7 @@ pub fn Table(props: TableProps) -> Element {
                         let val = e.value();
                         search_query.set(val.clone());
                         page.set(0);
+                        #[cfg(target_arch = "wasm")]
                         update_search_param(&val);
                     }
                 }

